resource/resource.go

```diff
@@ -335,8 +335,10 @@ func repodirMatches(dir, owner, repo string) error {
 	right := []string{gu.Host, gu.Owner, gu.Repo}
 	for i, l := range left {
 		r := right[i]
-		if strings.ToLower(l) != strings.ToLower(r) {
-			return fmt.Errorf("remote: %v: got: %q; want: %q: %w", remote, r, l, errWrongRemote)
+		if !strings.EqualFold(l, r) {
```

Comment:
could you please add a test ?

Reply:
I thought this was already covered here:
https://github.com/Pix4D/cogito/blob/04b201ef9f445fb53ec8e551d5a87f130a266a1f/resource/resource_test.go#L351
and it seems that chaining `strings.ToLower(l) != strings.ToLower(r)` to `!strings.EqualFold(l, r)` works fine,
but the problem is my new error message, which corresponds only to the last check (repo name), but not the first two (github url, user/organization)

Reply:
I have "added" an extra test and corrected the code, but I don't like how I did it. Can you give me an advice what should I do here? 
I mean given what you wrote in the second part of the ticket: https://pix4dbug.atlassian.net/browse/PCI-1937 you obviously tough that the error message was not clear enough here, so I want to improve it, but my original comment actually applies only when comparing the last item (repo name)

Reply:
> I thought this was already covered here:
You are right, I missed that.

Reply:
In my understanding, since there was already a test, no need to add another one.
If you still have questions, please ping me here.

Reply:
Yes, at the end I did not add any test. What I did for the test is changed the test name to point out that both can be correct (wrong SSH config but it can also be that the source configuration is wrong).

```
-  {"repo with wrong SSH remote", "a-repo", sshRemote("owner", "repo"), errWrongRemote},
+ {"repo with wrong SSH remote or wrong source config", "a-repo", sshRemote("owner", "repo"), errWrongRemote},
```

Reply:
ok